src/block-service.js

```diff
@@ -1,19 +1,12 @@
 'use strict'
 
-const Block = require('ipfs-block')
-const bl = require('bl')
 const async = require('async')
 
 // BlockService is a hybrid block datastore. It stores data in a local
 // datastore and may retrieve data from a remote Exchange.
 // It uses an internal `datastore.Datastore` instance to store values.
 function BlockService (ipfsRepo, exchange) {
-  this.addBlock = (block, callback) => {
-    const ws = ipfsRepo.datastore.createWriteStream(block.key, block.extension, callback)
-
-    ws.write(block.data)
-    ws.end()
-  }
+  this.addBlock = ipfsRepo.datastore.put
```

Comment:
This should probably be bound to be safe: `this.addBlock = ipfsRepo.datastore.put.bind(ipfsRepo.datastore)`